circle_suck/base/utils.py

```diff
@@ -11,3 +13,57 @@ def merge_dicts(*dicts):
     { 'a': 1, 'b': 2, 'c': 3 }
     """
     return dict(sum([d.items() for d in dicts], []))
+
+def find_all_connected(graph, minsize=2):
+	"""
+	Takes in a graph in the format: {footballteam1:[footballteam1 beat these footballteams]...}
+	Ex: {Cal:[Utah, Oregon]...}
+	
+	Returns a list of all disjoint cycles in the given graph (not sorted by length)
+	"""
+    listoflists = []
+    list = tarjan(graph)
+    for item in list[:]:
+        if len(item) <= minsize:
+            list.remove(item)
+    #print(list)
+    for con_comp in list:
+        newgraph = {}
+        for vertex, to in graph.items():
+            if vertex in con_comp:
+                newgraph[vertex] = to
+        listoflists.append(find_longest_cycle(newgraph))
+    return listoflists
+
+def find_all_cycles(graph, start):
+    def find_all_paths(graph, start, end, path=[]):
```

Comment:
you dont need to pass `graph` in again, since it has access to the parent's `graph` variable